drafts/19008-security.adoc

```diff
@@ -8,76 +8,103 @@ seo-title: Securing Open Liberty apps and microservices - OpenLiberty.io
 
 //
 
 == Example: Using OpenID Connect and JWTs to authenticate users and secure microservices:
 
-The following sample code and configurations use three servers to demonstrate authenticating a user and accessing a secured microservice using a JWT. The three pieces of this are:
+The following sample code and configurations use two servers to demonstrate authenticating a user, building a JWT and accessing a secured microservice using that JWT. The pieces of this are:
 
-* A server called `oidc_client` with the OpenID Connect Client feature (`openIdConnectClient-1.0`) hosts a servlet app secured with OpenID Connect.  
+* A server called `oidc_client` with the Social Login feature (`socialLogin-1.0`) hosts a servlet app secured with OpenID Connect. Google is the designated security provider.  
 +
 * A server called `microservice` with the MicroProfile JWT feature (`mpJwt-1.1`) hosts a JAX-RS microservice that is secured with MicroProfile JWT.
-+
-* A server called `oidc_provider` with the OpenID Connect Server feature (`openidConnectServer-1.0`) provides user authentication for the OpenID Connect client.
 
-When a user attempts to access the servlet, their browser is redirected to the OpenID Connect Provider for them to enter their ID and password.  Upon success, the browser gets redirected back to the client with an authorization code.  The client then contacts the provider, obtains a JWT access token, then uses it to create an authenticated user on the client, which then gets access to the servlet.
+An application was first registered with Google, which then supplied the client ID and client secret used in server.xml.
+(Registration instructions are https://developers.google.com/identity/protocols/OAuth2[here] ).
+
+The public certificate from Google was added to the client, and certificates exchanged between client and microservice as described https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/twlp_add_trust_cert.html[here].
+
+When a user attempts to access the servlet, their browser is redirected to Google for them to enter their ID and password.  Upon success, the browser gets redirected back to the client with an authorization code.  The client then contacts the provider (Google) on a back channel, obtains an access token, then uses it to create an authenticated user on the client, which then gets access to the servlet.
 
-The servlet then makes a REST call to the microservice.  REST clients can be configured to automatically propagate an access token if they have one, so the JWT access token gets sent along in the request. The microservice server is configured to trust JWTs issued by the provider, so it grants access to the JAX-RS service.
+The servlet then creates a JWT and makes a REST call to the microservice.  The JWT gets sent along in an HTTP header. The microservice server is configured to trust JWTs issued by oidc_client, so it grants access to the JAX-RS service.
```

Comment:
"...The `microservice` server is configured to trust JWTs issued by the `oidc_client`, which hosts the servlet..."  ? (I found I had to check back to double-check which was which.)